### ISSUE TITLE
[BUG] Getting `avoid-collection-methods-with-unrelated-types` error for a valid code

### ISSUE THREAD

**Issue by saileshbro** [dart-code-checker/dart-code-metrics#1204]
### Environment and configuration

DCM version: ^5.6.0
Dart SDK version: 2.19.3

<details>
<summary>Configuration</summary>

```yaml
include: package:very_good_analysis/analysis_options.yaml

dart_code_metrics:
  metrics-exclude:
    - test/**
    - "**/*.router.dart"
    - "**/*.gen.dart"
    - "**/*.locator.dart"
    - "**/*.freezed.dart"
    - "**/*.config.dart"
    - "**/*.logger.dart"
    - "**/*.form.dart"
    - "**/*.mocks.dart"
    - "**/*.g.dart"
    - "**/firebase_options*.dart"
  rules:
    - no-boolean-literal-compare
    - prefer-trailing-comma
    - no-equal-then-else
    - always-remove-listener:
        severity: warning
    - avoid-returning-widgets
    - prefer-const-border-radius:
        severity: warning
    - prefer-correct-edge-insets-constructor:
        severity: warning
    - avoid-collection-methods-with-unrelated-types
    - avoid-duplicate-exports
    - avoid-global-state
    - avoid-missing-enum-constant-in-map
    - avoid-redundant-async
    - avoid-unnecessary-type-casts:
        severity: warning
    - avoid-unrelated-type-assertions
    - avoid-unused-parameters:
        severity: warning
    - binary-expression-operand-order
    - double-literal-format
    - prefer-immediate-return:
        severity: warning
    - prefer-iterable-of:
        severity: warning
    - prefer-last:
        severity: warning

analyzer:
  plugins:
    - dart_code_metrics
  errors:
    missing_required_param: error
    missing_return: error
    must_be_immutable: error
    parameter_assignments: error
    sort_pub_dependencies: ignore
    sort_unnamed_constructors_first: warning
    avoid_renaming_method_parameters: warning
    unnecessary_getters_setters: ignore
    directives_ordering: ignore
    todo: ignore
    library_private_types_in_public_api: ignore
    no_leading_underscores_for_local_identifiers: ignore
    public_member_api_docs: ignore
    omit_local_variable_types: ignore
    lines_longer_than_80_chars: ignore
    unawaited_futures: ignore
    unnecessary_lambdas: warning
    avoid_equals_and_hash_code_on_mutable_classes: warning
    avoid_positional_boolean_parameters: ignore
    use_setters_to_change_properties: ignore
    prefer_constructors_over_static_methods: warning

  exclude:
    - "**/*.router.dart"
    - "**/*.gen.dart"
    - "**/*.locator.dart"
    - "**/*.freezed.dart"
    - "**/*.config.dart"
    - "**/*.logger.dart"
    - "**/*.form.dart"
    - "**/*.mocks.dart"
    - "**/*.g.dart"

```
</details>

### What did you do?

```dart
  final List<String> _selectedFriends = [];

  bool get isFriendsSelected => _selectedFriends.isNotEmpty;

  bool isSelected(Friend friend) => _selectedFriends.contains(friend.ref.id);

  void onFriendSelected(Friend friend) {
    final value = !isSelected(friend);
    if (value) {
      _selectedFriends.add(friend.ref.id);
    } else {
      _selectedFriends.remove(friend.ref.id);
    }
    notifyListeners();
  }
```

### What did you expect to happen?

I am getting a linting error for `Avoid collection methods with unrelated types.`, with a valid dart code. It should not show the linting error for this piece of code.
<img width="949" alt="image" src="https://user-images.githubusercontent.com/43571990/223425124-c286bc5e-d7f7-4232-bd0e-cfca47f1b121.png">

### What actually happened?

```
Avoid collection methods with unrelated types.
dart[avoid-collection-methods-with-unrelated-types](https://dcm.dev/docs/individuals/rules/common/avoid-collection-methods-with-unrelated-types)
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by incendial** [comment]
@saileshbro can you hover on a id field and show the type?

**Comment by saileshbro** [comment]
@incendial `id` is `String`
<img width="974" alt="image" src="https://user-images.githubusercontent.com/43571990/223435645-1483a42a-6879-4b00-91d9-45723c6b2456.png">

**Comment by saileshbro** [comment]
@incendial I think this happens specially after i run `build_runner`

**Comment by incendial** [comment]
@saileshbro could you provide a repro?
